docker/Dockerfile_commcarehq_base

```diff
@@ -27,30 +22,28 @@ RUN wget -O jdk.tar.gz --quiet --no-check-certificate --no-cookies --header "Coo
  && update-alternatives --auto java \
  && rm -f jdk.tar.gz
 
-# prefer https for git checkouts made by pip
-RUN git config --global url."https://".insteadOf git://
-
-RUN pip install --upgrade pip
+COPY requirements/requirements.txt \
+     requirements/dev-requirements.txt \
+     requirements/test-requirements.txt \
+     package.json \
+     /vendor/
 
-RUN mkdir /vendor
-ENV PYTHONUSERBASE /vendor
-
-COPY requirements/requirements.txt /vendor/requirements.txt
-COPY requirements/dev-requirements.txt /vendor/dev-requirements.txt
-COPY requirements/test-requirements.txt /vendor/test-requirements.txt
-
-RUN pip install \
+# prefer https for git checkouts made by pip
+RUN git config --global url."https://".insteadOf git:// \
+ && pip install --upgrade pip \
+ && pip install \
     -r /vendor/requirements.txt \
     -r /vendor/dev-requirements.txt \
     ipython \
     --user --upgrade \
-    && rm -rf /root/.cache/pip
-
-COPY package.json /vendor/package.json
+ && rm -rf /root/.cache/pip
 
-RUN cd /vendor \
-    && npm shrinkwrap \
-    && npm -g install \
-    && npm cache clean
-
-ENV PATH=/vendor/bin:$PATH
+RUN npm -g install \
+    bower \
+    grunt-cli \
+    uglify-js \
+ && echo '{ "allow_root": true }' > /root/.bowerrc \
+ && cd /vendor \
+ && npm shrinkwrap \
+ && npm -g install \
+ && npm cache clean
```

Comment:
I think grouping `npm -g install bower ...` in the same `RUN` with `npm cache clean` cuts down the size of the final image.